### PR TITLE
Add build time to deployments

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

This PR adds the build time to deployments for easier access

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by TorstenDittmann — COMMENTED** [review]
Can we extend the existing tests and also check if this needs to be added to the functions worker?

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -832,6 +832,7 @@
             $result->setAttribute('status', $build->getAttribute('status', 'processing'));
             $result->setAttribute('buildStderr', $build->getAttribute('stderr', ''));
             $result->setAttribute('buildStdout', $build->getAttribute('stdout', ''));
+            $result->setAttribute('buildTime', $build->getAttribute('duration', 0));
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply app/controllers/api/functions.php:835]: They are both correct, The deployment just gets the buildTime by checking how long the build took. We can rename the buildTime to duration if you like

File: app/workers/builds.php [hunk 2]
```diff
@@ -181,6 +181,10 @@ protected function buildDeployment(Document $project, Document $function, Docume
             $build->setAttribute('stderr', $response['stderr']);
             $build->setAttribute('stdout', $response['response']);
 
+            /* Also update the deployment buildTime */
+            $deployment->setAttribute('buildTime', $response['duration']);
+            $deployment = $dbForProject->updateDocument('deployments', $deployment->getId(), $deployment);
```
Comment (TorstenDittmann) [inline app/workers/builds.php:186]: we don't need to update this right?

Comment (PineappleIOnic) [reply app/workers/builds.php:186]: I added this to help with events not receiving buildTime, if you don't mind about events getting it, we can remove it

Comment (TorstenDittmann) [reply app/workers/builds.php:186]: yeah, but do we need to update the document?

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**